src/components/Account/ScreenConnected.js

```diff
@@ -36,108 +36,131 @@ function AccountScreenConnected({ onClosePopover, wallet }) {
               align-items: center;
-              justify-self: flex-end;
-              padding: ${0.5 * GU}px;
-              &:active {
-                background: ${theme.surfacePressed};
-              }
+              margin-right: ${3 * GU}px;
             `}
           >
-            <IdentityBadge
-              entity={wallet.account}
-              compact
-              badgeOnly
-              css="cursor: pointer"
-            />
-            <IconCopy
+            <img
+              src={providerInfo.image}
+              alt=""
               css={`
-                color: ${theme.hint};
+                width: ${2.5 * GU}px;
+                height: ${2.5 * GU}px;
+                margin-right: ${0.5 * GU}px;
+                transform: translateY(-2px);
               `}
             />
-          </ButtonBase>
+            <span>
+              {providerInfo.id === 'unknown' ? 'Wallet' : providerInfo.name}
+            </span>
+          </div>
+          <div
+            css={`
+              display: flex;
+              align-items: center;
+              justify-content: flex-end;
+              width: 100%;
+            `}
+          >
+            <ButtonBase
+              onClick={() => copy(wallet.account)}
+              focusRingRadius={RADIUS}
+              css={`
+                display: flex;
+                align-items: center;
+                justify-self: flex-end;
+                padding: ${0.5 * GU}px;
+                &:active {
+                  background: ${theme.surfacePressed};
+                }
+              `}
+            >
+              <IdentityBadge
+                entity={wallet.account}
+                compact
+                badgeOnly
+                css="cursor: pointer"
+              />
+              <IconCopy
+                css={`
+                  color: ${theme.hint};
+                `}
+              />
+            </ButtonBase>
+          </div>
         </div>
-      </div>
-      <div
-        css={`
-          padding: ${2 * GU}px 0;
-          border-bottom: 1px solid ${theme.border};
-        `}
-      >
         <div
           css={`
-            display: flex;
-            align-items: center;
-            color: ${theme.positive};
-            ${textStyle('label2')};
+            padding: ${2 * GU}px 0;
           `}
         >
-          <IconCheck size="small" />
-          <span
+          <div
             css={`
-              margin-left: ${0.5 * GU}px;
+              display: flex;
+              align-items: center;
+              color: ${theme.positive};
+              ${textStyle('label2')};
             `}
           >
-            {`Connected to Ethereum ${walletNetworkName} Network`}
-          </span>
+            <IconCheck size="small" />
+            <span
+              css={`
+                margin-left: ${0.5 * GU}px;
+              `}
+            >
+              {`Connected to Ethereum ${walletNetworkName} Network`}
```

Comment:
i think that i already pointed this in a previous pr, but we might want to remove the Ethereum when we are on xdai

Reply:
Good catch, maybe we could remove `Ethereum` entirely and just leave the rest?

Reply:
yeah it makes sense